lib/unifex/code_generator.ex

```diff
@@ -12,26 +12,44 @@ defmodule Unifex.CodeGenerator do
   @doc """
   Generates boilerplate code using generator implementation from `Unifex.CodeGenerators`.
   """
-  @spec generate_code(Specs.t()) :: {header :: code_t, source :: code_t}
+  @spec generate_code(Specs.t()) :: [{header :: code_t, source :: code_t}]
   def generate_code(specs) do
-    generator = get_generator(specs)
-    header = generator.generate_header(specs)
-    source = generator.generate_source(specs)
-    {header, source}
+    generators = get_generators(specs)
+
+    generators
+    |> Enum.map(fn {interface, generator} ->
+      header = generator.generate_header(specs)
+      source = generator.generate_source(specs)
+      {header, source, interface}
+    end)
   end
 
-  defp get_generator(%Specs{name: name, interface: nil}) do
+  defp get_generators(%Specs{name: name, interface: nil}) do
     {:ok, bundlex_project} = Bundlex.Project.get()
+    config = bundlex_project.config
+
+    interfaces = [:natives, :libs] |> Enum.find_value(&get_in(config, [&1, name, :interface]))
 
-    [:nifs, :cnodes]
-    |> Enum.find(&(bundlex_project.config |> Keyword.get(&1, []) |> Keyword.has_key?(name)))
-    |> case do
-      :nifs -> Unifex.CodeGenerators.NIF
-      :cnodes -> Unifex.CodeGenerators.CNode
+    case interfaces do
+      [] -> raise "Interface for native #{name} is not specified.
+        Please specify it in your *.spec.exs or bundlex.exs file."
+      _ -> interfaces |> Enum.map(&get_generator_module_name(&1))
     end
   end
 
-  defp get_generator(%Specs{interface: interface}) do
-    Module.concat(Unifex.CodeGenerators, interface)
+  defp get_generators(%Specs{interface: interfaces}) do
+    interfaces |> Bunch.listify() |> Enum.map(&get_generator_module_name(&1))
+  end
+
+  defp get_generator_module_name(interface) do
+    module_name =
+      case interface do
+        :nif -> :NIF
+        :cnode -> :CNode
+        other -> raise "Valid interfaces are :nif and :cnode. Passed #{other}"
```

Comment:
Interface handling in Unifex is designed to allow implementing custom interfaces by implementing `Unifex.CodeGenerators.*` modules. That approach makes that impossible